### ISSUE TITLE
AutoFocus crash on Sony Xperia D5503

### ISSUE THREAD

**Issue by JerabekJakub** [dm77/barcodescanner#79]
Hi, I've received crash report with this stack trace:

java.lang.RuntimeException: autoFocus failed
       at android.hardware.Camera.native_autoFocus(Camera.java)
       at android.hardware.Camera.autoFocus(Camera.java:1120)
       at me.dm7.barcodescanner.core.CameraPreview$1.run(SourceFile:237)
       at android.os.Handler.handleCallback(Handler.java:733)
       at android.os.Handler.dispatchMessage(Handler.java:95)
       at android.os.Looper.loop(Looper.java:212)
       at android.app.ActivityThread.main(ActivityThread.java:5135)
       at java.lang.reflect.Method.invokeNative(Method.java)
       at java.lang.reflect.Method.invoke(Method.java:515)
       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:877)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:693)
       at dalvik.system.NativeStart.main(NativeStart.java)

Android version is 4.4.4 and I used the latest version (1.7) of ZBar. It's crashing repeatedly.

**Comment by wyemun** [comment]
Confirmed. It happens mostly on Sony Xperia Z series.

```java
java.lang.RuntimeException: autoFocus failed
at android.hardware.Camera.native_autoFocus(Native Method)
at android.hardware.Camera.autoFocus(Camera.java:1120)
```

**Comment by Neristance** [comment]
Hey There,

I can also confirm that it happens on the Z1, autofocus works if you disable it when starting and later enable it again when the Preview is running

**Comment by andressantiago** [comment]
Neristance, can you elaborate? Should I be calling .setAutoFocus(false) in my onPause in the scanner activity, then .setAutoFocus(true) on surfaceCreated in CameraPreview.java?

**Comment by Neristance** [comment]
Sorry for the late answer i did it like that in onResume: 
scannerView.setAutoFocus(false);
scannerView.startCamera();
scannerView.setAutoFocus(false);

And i have overridden the ZXingScannerView class that it calls
setAutoFocus(true)
at the end of onPreviewFrame

I hope it helps

**Comment by plastiv** [comment]
I believe autofocus crashes due to missing boolean check (See #7 and #41). 
Please check my drop-in replacement which contains fix and let us know if it helps. I believe the author of the lib would appreciate the sample with the issue or field report to confirm that bug is fixed. 

```
repositories {
    maven { url "https://raw.github.com/plastiv/barcodescanner/mvn_repo" }
}
dependencies {
    compile 'io.github.plastiv.barcodescanner:zxing:1.7'
    compile 'io.github.plastiv.barcodescanner:zbar:1.7'
}
```

**Comment by dm77** [comment]
#41 is now merged in.

**Comment by dm77** [comment]
Don't think this is resolved, at least according to #85
